src/App.tsx

```diff
@@ -1,33 +1,18 @@
 import type { FC } from 'react'
-import React, { useState, useCallback } from 'react'
-import reactLogo from './assets/react.svg'
-import './App.scss'
+import React, { useCallback } from 'react'
+import { useAppStore } from './store/appStore'
 
 const App: FC = () => {
-  const [count, setCount] = useState(0)
-
-  const handleClick = useCallback(() => {
-    setCount(count => count + 1)
-  }, [])
-
+  const theme = useAppStore(store => store.theme)
+  const switchTheme = useAppStore(store => store.switchTheme)
+  const handleTheme = useCallback(() => {
+    switchTheme()
+  }, [switchTheme])
```

Comment:
I'd remove this function. You could just pass `switchTheme` directly to the `onClick` handler (was the meaning of my previous comment) without any impact, since Zustand internally memoizes the resulting selector function, afaik.

Reply:
Oops.. so true 😱